lib/puppet/provider/selinux_state/selinux_state.rb

```diff
@@ -18,5 +19,13 @@ def ensure=(should)
           setenforce "0"
       end
     end
+
+    # If we're going from disabled to enabled, we should touch /.autorelabel
+    relabel_cases = ['enforcing','permissive',:true]
+    if resource[:autorelabel]
+      if (relabel_cases.include? should) and ('disabled'.eql? self.ensure)
```

Comment:
* Use `&&` instead of `and`.
* `permissive` -> `enforcing` is also a relabel-worthy case
  * Basically, anything that is `off` to `on` is relabel-worthy

Reply:
Doesn't permissive count as on?

Reply:
@jeefberkey No. You can have SELinux contexts change while in permissive mode that will break when you get into enforcing mode. It is *much less common*, but it can 100% still happen.

Relevant link: https://access.redhat.com/documentation/en-US/Red_Hat_Enterprise_Linux/6/html/Security-Enhanced_Linux/sect-Security-Enhanced_Linux-Working_with_SELinux-Changing_SELinux_Modes.html (see the first note)

Reply:
👍 thanks for the link. Will implement